sphinx_gallery/gen_rst.py

```diff
@@ -351,6 +363,7 @@ def save_figures(image_path, fig_count, gallery_conf):
         current_fig = image_path.format(fig_count + fig_mngr.num)
         fig.savefig(current_fig, **kwargs)
         figure_list.append(current_fig)
+        fig._sg_captured = True
```

Comment:
Rather than this monkey-patching I think it would be slightly better to have an additional argument in save_figures, which is a set that remembers the already captured figures.

Alternatively we could have a matplotlib index and a mayavi index keeping track of the number of the last plot saved.